app/instrumentation-tests/src/org/commcare/tests/CaseClaimTest.java

```diff
@@ -0,0 +1,189 @@
+        installApp(appName, cczName);
+    }
+
+    @Test
+    public void testCaseClaimByDifferentUser() {
+        String name = "cordelia";
+        String location = "boston";
+
+        // Make sure we close off any existing cases due to previous test failures.
+        HQApi.closeExistingCases(name, "human", "d58f7a55dbe2bf22d0b6838311ada205");
+
+        // Waiting here cuz, HQ sometimes is out of sync and might give stale data.
+        onView(isRoot()).perform(Utility.sleep(TimeUnit.SECONDS.toMillis(15)));
+
+        Utility.login("claim_test1", "123");
+
+        // Make sure that a case with cordelia doesn't exist.
+        onView(withText("Start"))
+                .perform(click());
+        onView(withText("Follow Up"))
+                .perform(click());
+        onView(withText("SEARCH ALL CASES"))
+                .perform(click());
+        onView(Utility.find(
+                allOf(withClassName(endsWith("EditText"))),
+                1
+        )).perform(typeText(name));
+        closeSoftKeyboard();
+        onView(withId(R.id.request_button))
+                .perform(click());
+        onView(withId(R.id.request_button))
+                .check(matches(isDisplayed())); // The request doesn't do anything which confirms case doesn't exists.
+
+        // Register a new case with name cordelia
+        pressBackUnconditionally();
+        pressBackUnconditionally();
+        onView(withText("Registration"))
+                .perform(click());
+        onView(withClassName(endsWith("EditText")))
+                .perform(typeText(name));
+
+        onView(withId(R.id.nav_btn_next))
+                .perform(click());
+
+        onView(withClassName(endsWith("EditText")))
+                .perform(typeText(location));
+        onView(withId(R.id.nav_btn_finish))
+                .perform(click());
+
+        // Confirm newly created case can be claimed.
+        onView(withText("Start"))
+                .perform(click());
+        onView(withText("Follow Up"))
+                .perform(click());
+        onView(withText(name))
+                .check(matches(isDisplayed()));
+        onView(withText(name))
+                .perform(click());
+        onView(withText("Continue"))
+                .perform(click());
+        onView(withText("Close"))
+                .check(matches(isDisplayed()));
+
+        // Login with another user.
+        pressBackUnconditionally();
+        pressBackUnconditionally();
+        pressBackUnconditionally();
+        pressBackUnconditionally();
+        Utility.logout();
+
+        onView(isRoot()).perform(Utility.sleep(TimeUnit.SECONDS.toMillis(15)));
+
+        Utility.login("claim_test2", "123");
+        // Syncing once again to pull all the case data.
+        onView(withText("Sync with Server"))
+                .perform(click());
+
+        // Somehow, espresso starts before data pull is completed.
+        onView(isRoot()).perform(Utility.sleep(TimeUnit.SECONDS.toMillis(15)));
```

Comment:
I guess it's not possible to perform anything with progress bar in espresso.

Reply:
It's weird in that case that espresso starts even before the data pull as seems like it waits by itself for the async tasks to be executed which is the reason we can't perform operations on progress bar right ? so both these behaviours looks contradicting to me with possibility of only one to exist. If they are both co-existing it would be nice to see what exactly is going on here.

Reply:
yeah I confused myself and added 2 waits there. I just wanted to wait for HQ since it gave me wrong data many times. Updated the code to remove these waits.

Reply:
sounds good.